File: app/src/main/java/com/comm/util/dagger/dn2/MainActivity4.java
package com.comm.util.dagger.dn2;

import javax.inject.Inject;

import android.os.Bundle;
import android.util.Log;
import androidx.appcompat.app.AppCompatActivity;
import com.comm.util.R;

public class MainActivity4 extends AppCompatActivity {
    String TAG = "MainActivity3";
    @Inject
    User4 user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main3);
        //DaggerApp
        DaggerApplicationComponent4.create().inject(this);
        Log.i(TAG, "onCreate: user= " + user);
    }

}

File: app/src/main/java/com/comm/util/dagger/dn2/MainActivity4.kt
package com.comm.util.dagger.dn2

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.comm.util.R
import com.comm.util.dagger.dn2.di.DaggerApplicationComponent4
import com.comm.util.dagger.dn2.di.User4
import javax.inject.Inject

class MainActivity4 : AppCompatActivity() {
    var TAG = javaClass.simpleName

    @Inject
    lateinit var user: User4
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main3)
        DaggerApplicationComponent4.create().inject(this)
        Log.i(TAG, "onCreate: user= $user")
    }
}